File: static/code/sample/tethered/gdl-segas-b.cs
// In this sample:
// Warm up the sensor for 10 seconds, then read the analog value

using GHIElectronics.DUELink;

var availablePort = DUELinkController.GetConnectionPort();
var duelink = new DUELinkController(availablePort);

void Heater(bool value) {
    var v = value ? 1 : 0;
    duelink.Engine.ExecuteCommand($"Heater({v})");
}

float SensorRead() {
    var ret = duelink.Engine.ExecuteCommand("SenRead()"); 
    return ret;
}

var counter = 0;

while (true) {
    Thread.Sleep(1000);

    // Wait for heat on for 10 seconds
    if (counter == 0) {
        Heater(true);
    }
    counter++;

    if (counter < 10) {
        Console.WriteLine($"Wait for {counter}/10 seconds");
        continue;
    }

    var volt = SensorRead();

    Console.WriteLine($"Volt: {volt}");
}

File: static/code/sample/tethered/gdl-segas-b.py
# In this sample:
# Warm up the sensor for 10 seconds, then read the analog value

import time
from DUELink.DUELinkController import DUELinkController

availablePort = DUELinkController.GetConnectionPort()
duelink = DUELinkController(availablePort)

# method
def Heater(value: bool):
    v = 1 if value else 0
    duelink.Engine.ExecuteCommand(f"Heater({v})")

def SensorRead() -> float:
    ret = duelink.Engine.ExecuteCommand("SenRead()")
    return ret

counter = 0

while True:
    time.sleep(1)  # Equivalent to Thread.Sleep(1000)

    # Wait for heat on for 10 seconds
    if counter == 0:
        Heater(True)
    counter += 1

    if counter < 10:
        print(f"Wait for {counter}/10 seconds")
        continue

    volt = SensorRead()
    print(f"Volt: {volt}")
